src/query/service/src/interpreters/interpreter_table_drop.rs

```diff
@@ -110,13 +110,14 @@ impl Interpreter for DropTableInterpreter {
         // but the table still exists, in the interval maybe some unexpected things will happen.
         // drop the ownership
         let role_api = UserApiProvider::instance().get_role_api_client(&self.plan.tenant)?;
-        role_api
-            .revoke_ownership(&OwnershipObject::Table {
-                catalog_name: self.plan.catalog.clone(),
-                db_id: db.get_db_info().ident.db_id,
-                table_id: tbl.get_table_info().ident.table_id,
-            })
-            .await?;
+        let owner_object = OwnershipObject::Table {
+            catalog_name: self.plan.catalog.clone(),
+            db_id: db.get_db_info().ident.db_id,
+            table_id: tbl.get_table_info().ident.table_id,
+        };
+        let role = role_api.get_ownership(&owner_object).await?.map(|o| o.role);
+
+        role_api.revoke_ownership(&owner_object, role).await?;
```

Comment:
is it possible to put the get_ownership part inside the `revoke_ownership()` call?

like replacing:

```
        let role = role_api.get_ownership(&owner_object).await?.map(|o| o.role);

        role_api.revoke_ownership(&owner_object, role).await?;
```

with:

```
        role_api.revoke_ownership(&owner_object).await?;
```

Reply:
this may help reduce some duplication on keeping the calling invariance consistent.

Reply:
Done